python/docs/source/user_guide/arrow_pandas.rst

```diff
@@ -410,3 +410,11 @@ described in `SPARK-29367 <https://issues.apache.org/jira/browse/SPARK-29367>`_
 ``pandas_udf``\s or :meth:`DataFrame.toPandas` with Arrow enabled. More information about the Arrow IPC change can
 be read on the Arrow 0.15.0 release `blog <https://arrow.apache.org/blog/2019/10/06/0.15.0-release/#columnar-streaming-protocol-change-since-0140>`_.
 
+Setting Arrow self_destruct for memory savings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since Spark 3.2, the Spark configuration ``spark.sql.execution.arrow.pyspark.selfDestruct.enabled`` can be used to enable PyArrow's ``self_destruct`` feature, which can save memory when creating a Pandas dataframe via ``toPandas`` by freeing Arrow-allocated memory while building the Pandas dataframe.
+This option is experimental, and some operations may fail on the resulting Pandas dataframe due to immutable backing arrays.
+Typically, you would see the error ``ValueError: buffer source array is read-only``.
+Newer versions of Pandas may fix these errors by improving support for such cases.
+Additionally, this conversion may be slower because it is single-threaded.
```

Comment:
@lidavidm it would be great if we link related issues (so we can remove these notes later when they are solved).

Reply:
Unfortunately there's no real umbrella issue; it's an artifact of Pandas' implementation that's slowly being corrected as people find more cases.

Reply:
Hm, can we maybe file an issue in pandas side and like it here?

Reply:
We could link to https://github.com/pandas-dev/pandas/issues/33001 perhaps? It's not a blanket umbrella issue but it does cover a few hundred test cases and explains the issues facing a universal fix on their side.

Reply:
I doubt if that covers the current case here .. our issue happens specifically when the pandas DataFrame is from PyArrow, right?

I wonder if we can just show a reproducible example and file an issue in pandas dev side to track .. otherwise, this warning will likely stay forever ..

I am okay with as-is too .. assuming that we're sure on tracking the related issue.

Reply:
No, it's actually anytime that the underlying array is read-only. PyArrow is a common source of this but not the only source.